File: Bridge/Resources/Boolean.js
// @source Boolean.js

var boolean = {
    tryParse: function (s, v) {
        v.v = false;
        if (s === "False" || s === "false") {
	          v.v = false;
            return true;
        }
        if (s === "True" || s === "true") {
          v.v = true;
	        return true;
        }
        return false;
    }
}


Bridge.Boolean = boolean;

File: Bridge/System/Boolean.cs
using Bridge;

namespace System
{
    [External]
    [Name("Boolean")]
    //[Constructor("!!")]
    public struct Boolean : IComparable<Boolean>
    {
        [Template("!!")]
        public Boolean(object value)
        {
        }

        [Template( "Bridge.Boolean.tryParse({s}, {result})" )]
        public static bool TryParse( string s, out bool result )
        {
            result = false;
            return false;
        }

        [Template( "{value} === {other} ? 0 : {this} && !{other} ? 1 : -1" )]
        public int CompareTo( Boolean other ) {
           return 0;
        }
    }
}